src/optimizer/plugins/CombineCodingAndQuantityValuesOptimizer.ts

```diff
@@ -53,6 +53,15 @@ export default {
       const rules: fshrules.Rule[] = def.rules; // <-- this assignment makes TypeScript happier in the next chunk of code
       const typeCache: Map<string, fhirtypes.ElementDefinitionType[]> = new Map();
 
+      const ruleMap: { [path: string]: { [caretPath: string]: ExportableCaretValueRule[] } } = {};
+      rules
+        .filter(r => r instanceof ExportableCaretValueRule)
+        .forEach((r: ExportableCaretValueRule) => {
+          ruleMap[r.path] = ruleMap[r.path] || {};
```

Comment:
I think that in TypeScript, there is a preference for using nullish-coalescing `??` over `||` for this type of assignment. This is mainly to avoid overwriting "falsey" non-null values. This probably doesn't apply here, but since we're in TypeScript anyway, I think it's a good practice to use `??` anyway.